core/src/main/java/amino/run/compiler/PolicyStub.java

```diff
@@ -296,15 +297,22 @@ public String getMethodContent(MethodStub m, boolean isDMMethod) {
      * Returns the KernelRPC stub implementation code based on whether the method is a DM method or
      * Application method.
      *
+     * @param m
      * @param isDMMethod
      * @return Stub code for RPC call based on method type.
      */
-    public String getMethodRPCContent(boolean isDMMethod) {
-        if (isDMMethod == true) {
+    public String getMethodRPCContent(MethodStub m, boolean isDMMethod) {
+        if (m.retType.equals(Void.TYPE)) {
+            if (isDMMethod) {
+                return "$__makeKernelDMRPC($__method, $__params);";
```

Comment:
Too many returns in this <10lines method and also i think, coupling between caller/callee can be reduced. Can be optimized. How about this..?
Just change, `getMethodRPCContent`
`return "$__result = $__makeKernelRPC($__method, $__params);";` -> `return "$__makeKernelRPC($__method, $__params);";`
`return "$__result = $__makeKernelDMRPC($__method, $__params);";` -> `return "$__makeKernelDMRPC($__method, $__params);";`

And change getMethodContent as below:
![Selection_002](https://user-images.githubusercontent.com/35334869/60329328-c5e4aa00-99ad-11e9-8076-d707e67c87a4.png)